<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>732</width>
    <height>516</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>732</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout" columnstretch="1,9">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="0" column="1">
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="7,13">
        <item>
         <widget class="RoiView" name="roiScaleArea"/>
        </item>
        <item>
         <widget class="SpectrumView" name="spectrumArea">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <layout class="QVBoxLayout" name="verticalLayout" stretch="1,2,1,2,1">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QGridLayout" name="cntStatusGrid" rowstretch="1,1,1,1,1" columnstretch="1,2,2,2,1">
          <item row="2" column="2">
           <widget class="QLabel" name="stopChnAddr">
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="startChnCnt">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="label_11">
            <property name="text">
             <string>计数</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>终道</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QLabel" name="curChnAddr">
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="startChnAddr">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>道址</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>起道</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QLabel" name="totalCnt">
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>光标</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="4">
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="3" column="3">
           <widget class="QLabel" name="curChnCnt">
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="stopChnCnt">
            <property name="text">
             <string>0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>全谱</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QGridLayout" name="acqControlGrid" rowstretch="1,0,0,0,0" columnstretch="1,4,4,3,1">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item row="2" column="1">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>计数限制</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>仿真速度</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QDoubleSpinBox" name="simCountInput">
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="decimals">
             <number>0</number>
            </property>
            <property name="maximum">
             <double>100000.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>10000.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
            <property name="value">
             <double>10000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QDoubleSpinBox" name="simTimeInput">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>0.100000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="value">
             <double>100.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QDoubleSpinBox" name="simIntevalInput">
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>10.000000000000000</double>
            </property>
            <property name="stepType">
             <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
            </property>
            <property name="value">
             <double>2.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="2" colspan="2">
           <widget class="QComboBox" name="simMethodInput">
            <item>
             <property name="text">
              <string>固定时间</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>固定计数</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="4" column="2">
           <widget class="QLabel" name="simTimeElaspedLabel">
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>已采样</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>s</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>仿真方式</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>采样时间</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QComboBox" name="simTimeUnit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="sizeAdjustPolicy">
             <enum>QComboBox::AdjustToContents</enum>
            </property>
            <item>
             <property name="text">
              <string>s</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>m</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>h</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>次</string>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>ms</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="0">
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenFile"/>
   <addaction name="actionSaveFile"/>
   <addaction name="separator"/>
   <addaction name="actionStartAcq"/>
   <addaction name="actionStopAcq"/>
   <addaction name="actionClearAcq"/>
  </widget>
  <action name="actionOpenFile">
   <property name="icon">
    <iconset resource="mca.qrc">
     <normaloff>:/icons/toolbar/resources/open-archive.svg</normaloff>:/icons/toolbar/resources/open-archive.svg</iconset>
   </property>
   <property name="text">
    <string>打开能谱文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionStartAcq">
   <property name="icon">
    <iconset resource="mca.qrc">
     <normaloff>:/icons/toolbar/resources/play-button.svg</normaloff>:/icons/toolbar/resources/play-button.svg</iconset>
   </property>
   <property name="text">
    <string>开始采集</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionStopAcq">
   <property name="icon">
    <iconset resource="mca.qrc">
     <normaloff>:/icons/toolbar/resources/stop.svg</normaloff>:/icons/toolbar/resources/stop.svg</iconset>
   </property>
   <property name="text">
    <string>停止采集</string>
   </property>
   <property name="shortcut">
    <string>Return</string>
   </property>
  </action>
  <action name="actionClearAcq">
   <property name="icon">
    <iconset resource="mca.qrc">
     <normaloff>:/icons/toolbar/resources/clear.svg</normaloff>:/icons/toolbar/resources/clear.svg</iconset>
   </property>
   <property name="text">
    <string>清空能谱</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="actionSaveFile">
   <property name="icon">
    <iconset resource="mca.qrc">
     <normaloff>:/icons/toolbar/resources/save.svg</normaloff>:/icons/toolbar/resources/save.svg</iconset>
   </property>
   <property name="text">
    <string>保存能谱文件</string>
   </property>
   <property name="toolTip">
    <string>保存能谱文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>SpectrumView</class>
   <extends>QGraphicsView</extends>
   <header>spectrumview.h</header>
   <slots>
    <signal>signal1()</signal>
    <signal>simUpdated(uint64_t)</signal>
    <signal>spectrumUpdated()</signal>
    <signal>roiUpdated(PointsVector)</signal>
    <signal>cursorUpdated(uint64_t)</signal>
    <slot>startSimAcq(double)</slot>
    <slot>stopSimAcq()</slot>
    <slot>clearAcqData()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>RoiView</class>
   <extends>QGraphicsView</extends>
   <header>roiview.h</header>
   <slots>
    <slot>updateRoi(PointsVector)</slot>
    <slot>updateCursor(uint64_t)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mca.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionOpenFile</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_menu_openFileAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStartAcq</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_menu_startAcqAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStopAcq</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_menu_stopAcqAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClearAcq</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_menu_clearAcqDataAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumArea</sender>
   <signal>simUpdated(uint64_t)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_spectrum_simUpdated(uint64_t)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>367</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>28</x>
     <y>436</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumArea</sender>
   <signal>spectrumUpdated()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_spectrum_dataUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>458</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumArea</sender>
   <signal>roiUpdated(PointsVector)</signal>
   <receiver>roiScaleArea</receiver>
   <slot>updateRoi(PointsVector)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>73</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveFile</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_menu_saveFileAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumArea</sender>
   <signal>cursorUpdated(uint64_t)</signal>
   <receiver>roiScaleArea</receiver>
   <slot>updateCursor(uint64_t)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>536</x>
     <y>292</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spectrumArea</sender>
   <signal>cursorUpdated(uint64_t)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_spectrum_cursorUpdated(uint64_t)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>620</x>
     <y>376</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_menu_openFileAction()</slot>
  <slot>on_spectrum_simUpdated(uint64_t)</slot>
  <slot>on_menu_startAcqAction()</slot>
  <slot>on_spectrum_dataUpdated()</slot>
  <slot>on_menu_clearAcqDataAction()</slot>
  <slot>on_menu_saveFileAction()</slot>
  <slot>on_menu_stopAcqAction()</slot>
  <slot>on_spectrum_cursorUpdated(uint64_t)</slot>
 </slots>
</ui>
